core/client/src/main/java/alluxio/client/block/BlockStoreContext.java

```diff
@@ -313,11 +224,45 @@ public static void releaseNettyChannel(InetSocketAddress address, Channel channe
   }
 
   /**
+   * Acquires a block worker thrift client from the block worker thrift client pools.
+   *
+   * @param address the address of the block worker
+   * @return the block worker thrift client
+   * @throws IOException if it fails to connect to remote worker
```

Comment:
actually yes and that is the most likely (probably only) cause. I have improved the doc